app/src/main/java/net/tuxed/vpnconfigimporter/service/SerializerService.java

```diff
@@ -251,4 +274,182 @@ public JSONObject serializeDiscoveredAPI(DiscoveredAPI discoveredAPI) throws Unk
             throw new UnknownFormatException(ex);
         }
     }
+
+    /**
+     * Deserializes a JSON with a list of messages into an ArrayList of message object.
+     *
+     * @param jsonObject The JSON to deserialize.
+     * @return The message instances in a list.
+     * @throws UnknownFormatException Thrown if there was a problem while parsing.
+     */
+    public List<Message> deserializeMessageList(JSONObject jsonObject) throws UnknownFormatException {
+        try {
+            JSONObject dataObject = jsonObject.getJSONObject("data");
```

Comment:
Note, you can have the try/catch inside the loop, if one item fails it still uses other valid data.